spec/unit/product-manager.spec.js

```diff
@@ -357,4 +359,80 @@ describe('ProductManager', () => {
       ])
     })
   })
+
+  describe('ensure sameForAll constraint', () => {
+    const variants = [
+      {
+        sku: '1',
+        attributes: [
+          {
+            name: 'brandId',
+            value: '123'
+          },
+          {
+            name: 'isAvailable',
+            value: false
+          },
+          {
+            name: 'color',
+            value: 'red'
+          }
+        ]
+      },
+      {
+        sku: '2',
+        attributes: [
+          {
+            name: 'brandId',
+            value: '456'
+          },
+          {
+            name: 'isAvailable',
+            value: false
+          },
+          {
+            name: 'color',
+            value: 'green'
+          }
+        ]
+      }
+    ]
+
+    beforeEach(() => {
+      pM = new ProductManager(utils.logger, {})
+      pM.productTypeCache.set(productType.id, productType)
```

Comment:
ProductType object in `productType2.json` file does not have `id` field - it would be more correct to add it then work with `undefined` key

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/42/commits/523a653304d6d0bf5167f878a905dbfbd91a03db